xapian-bindings/go/xapian.go

```diff
@@ -0,0 +1,53 @@
+package xapian
```

Comment:
It doesn't seem ideal that this seems to require writing a wrapper for every wrapped function.  We use SWIG because it automates away all that drudgery.  We've had hand-coded Perl XS and Java JNI bindings in the past and they always ended up lagging behind the C++ API because of the extra work needed, whereas for SWIG-generated bindings in most cases a new method gets wrapped automatically, while new classes just need setting up according to what sort of classes they are (e.g. user-subclassable classes get special handling).

I'll take a look and see what we can do.

I think ideally SWIG itself would support a more Go-like way to wrap exceptions, but there doesn't seem to be much progress on that front.